packages/flame/lib/src/components/navigator.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:collection/collection.dart';
+import 'package:flame/src/components/component.dart';
+import 'package:flame/src/components/page.dart';
+
+class Navigator extends Component {
+  Navigator({
+    required Map<String, Page> pages,
```

Comment:
It would be nice if other types than strings could be used here too, enums for example, then the user can make their code a bit safer from runtime exceptions if they wish.

Reply:
Makes sense, however, I wanted to make an additional extension here, the `pageFactory` parameter. This would handle dynamically generated page names, for example `"level/357"`.

My thinking was that if you say `navigator.pushPage("level/357")`, then the navigator would split the name into two parts and then invoke `factories["level"].generatePage("357")` to create the page with the given name. But I don't see how this functionality could work with enums.